scenarios/cpp/windows/captioning/caption.cpp

```diff
@@ -0,0 +1,535 @@
+            }
+            else if (CancellationReason::Error == e.Reason)
+            {
+                std::ostringstream error;
+                error << "Encountered error.\n"
+                    << "ErrorCode: " << (int)e.ErrorCode << "\n"
+                    << "ErrorDetails: " << e.ErrorDetails << std::endl;
+                recognitionEnd.set_value(std::optional<std::string>{ error.str() }); // Notify to stop recognition.
+            }
+            else
+            {
+                std::ostringstream error;
+                error << "Request was cancelled for an unrecognized reason: " << (int)e.Reason << std::endl;
+                recognitionEnd.set_value(std::optional<std::string>{ error.str() }); // Notify to stop recognition.
+            }
+        });
+
+    speechRecognizer->SessionStopped.Connect([&userConfig, &recognitionEnd](const SessionEventArgs& e)
+        {
+            WriteToConsole("Session stopped.\n", userConfig);
+            recognitionEnd.set_value(std::nullopt); // Notify to stop recognition.
+        });
+
+    // Starts continuous recognition. Uses StopContinuousRecognitionAsync() to stop recognition.
+    speechRecognizer->StartContinuousRecognitionAsync().get();
+
+    // Waits for recognition end.
+    std::optional<std::string> result = recognitionEnd.get_future().get();
+
+    // Stops recognition.
+    speechRecognizer->StopContinuousRecognitionAsync().get();
+    
+    return result;
+}
+
+int main(int argc, char* argv[])
+{
+    const std::string usage = "Usage: caption.exe [-d] [-f] [-h] [-i file] [-l] [-o file] [-p phrases] [-q] [-r number] [-s] [-t] [-u] <subscriptionKey> <region>\n"
+    "              -d: Enable dictation mode.\n"
+    "              -f: Enable profanity filter.\n"
+    "              -h: Show this help and stop.\n"
+    "              -i: Input audio file *file* (default input is from the microphone.)\n"
+    "    -l languages: Enable language identification for specified *languages*.\n"
+    "                  Example: en-US,ja-JP\n"
+    "         -o file: Output to *file*.\n"
+    "      -p phrases: Add specified *phrases*.\n"
+    "                  Example: Constoso;Jessie;Rehaan\n"
+    "              -q: Suppress console output (except errors).\n"
+    "       -r number: Set stable partial result threshold to *number*.\n"
+    "                  Example: 3\n"
+    "              -s: Emit SRT (default is WebVTT.)\n"
+    "              -t: Enable TrueText.\n"
+    "              -u: Emit partial results instead of finalized results.\n";
+
+    try
+    {
+        if (CmdOptionExists(argv, argv + argc, "-h"))
+        {
+            std::cout << usage << std::endl;
+        }
+        else
+        {
+            UserConfig userConfig = UserConfigFromArgs(argc, argv);
+            std::shared_ptr<SpeechRecognizer> speechRecognizer = SpeechRecognizerFromUserConfig(userConfig);
+            Initialize(userConfig);
```

Comment:
Can this move within or right after `UserConfigFromArgs`? Currently it breaks up new SpeechRecognizer -> RecognizeContinuous flow.